bytes/src/main/scala/io/iohk/ethereum/utils/ByteStringUtils.scala

```diff
@@ -61,9 +61,14 @@ object ByteStringUtils {
   }
 
   def concatByteStrings(elements: Iterator[ByteStringElement]): ByteString = {
-    val builder = new mutable.ArrayBuilder.ofByte
-    elements.foreach(el => builder.addAll(el.asByteArray))
-    ByteString(builder.result())
+    val elementList: List[ByteStringElement] = elements.toList
+    val len = elementList.foldLeft(0)(_ + _.len)
+    val array = Array.ofDim[Byte](len)
+    elementList.foldLeft(0) { (index, el) =>
+      System.arraycopy(el.asByteArray, 0, array, index, el.len)
+      index + el.len
+    }
+    ByteString(array)
```

Comment:
Just a proposition, if we care about performance that much, and we know that underlying array will not be modified  we can use - ` ByteString.fromArrayUnsafe` which will reduce allocation even more